fsmonitor.c

```diff
@@ -50,17 +55,29 @@ int read_fsmonitor_extension(struct index_state *istate, const void *data,
 	}
 	istate->fsmonitor_dirty = fsmonitor_dirty;
 
+	assert(istate->fsmonitor_dirty->bit_size <= istate->cache_nr);
```

Comment:
Maybe make that a `BUG()`, too? I can easily imagine this to help stave off hard-to-debug problems in the future.